cmake/ccache.cmake

```diff
@@ -2,7 +2,7 @@
 # Distributed under the MIT software license, see the accompanying
 # file COPYING or https://opensource.org/license/mit/.
 
-if(NOT MSVC)
+if(CMAKE_GENERATOR MATCHES "Ninja|Makefiles")
```

Comment:
When building across different build directories, using the `base_dir` option helps work around absolute paths in `-I` compiler options, increasing the `ccache` cache hit rate.

Maybe, the word "effectiveness" is more appropriate, rather than "performance".

The change in the condition is not directly related to improving the hit rate. It enhances code correctness, as the method using of the `CMAKE_{C,CXX}_COMPILER_LAUNCHER` variables is applicable only to the mentioned generators.

Reply:
In commit "build: Improve `ccache` performance for different build directories" (66b73bbf375ff96bf4a77539c9cf3fe52fbc15d5)

Would be good to add a comment explaining `if(CMAKE_GENERATOR MATCHES "Ninja|Makefiles")` saying that COMPILER_LAUNCHER variable only works for these backends not other backends. Otherwise the check seems completely arbitrary and doesn't make sense.

Reply:
> Would be good to add a comment explaining `if(CMAKE_GENERATOR MATCHES "Ninja|Makefiles")` saying that COMPILER_LAUNCHER variable only works for these backends not other backends. Otherwise the check seems arbitrary and doesn't seem to make sense.

Thanks! Done.